modules/exploits/multi/http/mutiny_subnetmask_exec.rb

```diff
@@ -93,7 +93,7 @@ def on_new_session(session)
       %Q|sed -e s/NETMASK=.*/NETMASK=#{@netmask_eth0}/ ifcfg-eth0 > ifcfg-eth0.bak|,
       %Q|mv -f ifcfg-eth0.bak ifcfg-eth0|,
       %Q|/etc/init.d/network restart|
-    ] unless not @netmask_eth0
+    ] if @netmask_eth0
```

Comment:
Admittedly, this may be an attempt to double-negate to prevent a errant value when expecting a binary.....?

Reply:
I'm not sure I understand. It's checking truthiness.

Reply:
```
2.3.0 :001 > puts 'lies' if (true != nil && false != nil)
lies
 => nil 
2.3.0 :002 > puts 'lies' unless nil
lies
 => nil 
2.3.0 :003 > 
```

Reply:
That's what is meant by truthiness: it's checking whether a value is truthy or falsey, not `true` or `false`. Doesn't answer @bwatters-r7's implication.

```
[1] pry(main)> puts 'truthy' unless !(nil || false)
=> nil
[2] pry(main)> puts 'truthy' unless !('' || true)
(pry):2: warning: string literal in condition
truthy
=> nil
[3] pry(main)> puts 'truthy' if (nil || false)
=> nil
[4] pry(main)> puts 'truthy' if ('' || true)
(pry):4: warning: string literal in condition
truthy
=> nil
[5] pry(main)>
```

Reply:
It's actually @acammack-r7's implication via @bwatters-r7, so maybe Adam can explain the alleged edge case here.

Reply:
Not in Ruby code, in other times and places people have been worried about passing around non-boolean conditional values, but our conditionals expressions don't care. Sorry I gave the wrong impression @bwatters-r7.